### ISSUE TITLE
Rule proposal: avoid setup code outside of tests or hooks

### ISSUE THREAD

**Issue by lo1tuma** [lo1tuma/eslint-plugin-mocha#38]
This should be considered a warning:

```js
describe('foo',  function ()  {
    var fixture = fs.readFileSync('foo.txt');
});
```
You should read the file in a before hook instead. 

The problem with such code is that it will be executed during the process startup time. It will be even executed if you skip the test suite. 

_____

I'm not 100% sure how to achieve this. Disallowing every call expression would also disallow using forEach for parameterized tests.
